app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -150,23 +149,30 @@ def process_object_update_modify(obj, change_set, _missing_set = [])
     obj_ref  = obj._ref
 
     props = inventory_cache[obj_type][obj_ref].dup
```

Comment:
I think if you get rid of the `.dup` here we'll update the cache directly and you don't need the `remove_props` at all (can get rid of `update_inventory_cache` also
I only did the `.dup` to make sure that we _didn't_ update everything in the cache